File: simple-stack-example-basic-java-view/src/main/java/com/zhuinden/navigationexampleview/screens/HomeKey.java
package com.zhuinden.navigationexampleview.screens;

import com.google.auto.value.AutoValue;
import com.zhuinden.navigationexampleview.R;
import com.zhuinden.navigationexampleview.core.navigation.BaseKey;

/**
 * Created by Owner on 2017. 06. 29..
 */
@AutoValue
public abstract class HomeKey extends BaseKey {
    public static HomeKey create() {
        return new AutoValue_HomeKey();
    }

    @Override
    public int layout() {
        return R.layout.home_view;
    }
}

File: examples/simple-stack-example-basic-kotlin-view/src/main/java/com/zhuinden/navigationexamplekotlinview/screens/HomeKey.kt
package com.zhuinden.navigationexamplekotlinview.screens

import com.zhuinden.navigationexamplekotlinview.R
import com.zhuinden.navigationexamplekotlinview.core.navigation.BaseKey
import kotlinx.android.parcel.Parcelize

/**
 * Created by Owner on 2017. 06. 29..
 */
@Parcelize
data class HomeKey(private val placeholder: String = "") : BaseKey() {
    override fun layout(): Int = R.layout.home_view
}
